### DISCUSSION TITLE
Is it possible to connect to a RPis GPIO pin with gpiozero remotely WITHOUT changing a pins state when connecting?

### DISCUSSION THREAD

**Discussion by TRWoodland** [gpiozero/gpiozero#1102] [Q&A]
Is it possible to connect to a RPis GPIO pin with gpiozero remotely **WITHOUT** changing a pins state when connecting?

I have a relay connected to pin26.

When I use this command to connect:

`import gpiozero`
  
`from gpiozero.pins.pigpio import PiGPIOFactory`
  

  
`factory = PiGPIOFactory(host='PI IP ADDRESS')`
  
`pin = gpiozero.OutputDevice(26, active_high=False, initial_value=None, pin_factory=factory)`

&#x200B;

I have no problem:

* connecting
* Setting the active high/low state
* Setting the initial\_value to None so that when it connects it shouldn't change state.

The problem I have is that it very briefly resets the pin when connecting to it. 

So if the pin is high, and the relay active, when I remotely connect it will reset the pin then return it.

Please let me know if I've described the problem badly.

Thanks

**Comment by lurch** [comment]
I've just had a bit of a rummage in the gpiozero code, and it looks like when you construct an `OutputDevice` https://github.com/gpiozero/gpiozero/blob/master/gpiozero/output_devices.py#L72 it builds the `pin` object first, and _then_ sets the pin's function to output. And the `PiGPIOPin` constructor https://github.com/gpiozero/gpiozero/blob/master/gpiozero/pins/pigpio.py#L192 initially creates the pin with `self.factory.connection.set_mode(self._number, pigpio.INPUT)`, so I guess the pin being temporarily set as an input might be the cause of your brief reset.

AFAIK the only way of "fixing" this would be a big refactoring of https://gpiozero.readthedocs.io/en/latest/api_pins.html which doesn't really seem feasible, so I wonder if you might need to fix this by using https://abyz.me.uk/rpi/pigpio/python.html directly instead of using gpiozero? :shrug: 
ping @joan2937 in case they have any suggestions.
